src/state.ts

```diff
@@ -21,13 +21,85 @@ const workloadsLruCacheOptions = {
   updateAgeOnGet: false,
 };
 
-const state = {
+interface WorkloadAlreadyScanned {
+  namespace: string;
+  type: string;
+  uid: string;
+}
+
+interface WorkloadImagesAlreadyScanned {
+  namespace: string;
+  type: string;
+  uid: string;
+  imageIds: string[];
+}
+
+function getWorkloadAlreadyScannedKey(
+  workload: WorkloadAlreadyScanned,
+): string {
+  return `${workload.namespace}/${workload.type}/${workload.uid}`;
+}
+
+function getWorkloadImageAlreadyScannedKey(
+  workload: WorkloadAlreadyScanned,
+  imageId: string,
+): string {
+  return `${workload.namespace}/${workload.type}/${workload.uid}/${imageId}`;
```

Comment:
nit: you can re-use previous function
```suggestion
  return `getWorkloadAlreadyScannedKey(workload)/${imageId}`;
```

Reply:
Although it helps reuse code, IMO it doesn't help with readability since you have to go 2 levels deep to figure out what this key even contains